src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -1228,7 +1237,13 @@ public void PasteFromClipboard()
 
 				PasteFromClipboardPartial(clipboardText, selectionStart, selectionLength, currentText);
 
+#if __SKIA__
+				_suppressCurrentlyTyping = true;
+#endif
 				Text = currentText;
+#if __SKIA__
+				_suppressCurrentlyTyping = false;
+#endif
```

Comment:
This would need to be wrapped in a try/finally, in case setting the text fails for any reason.